action/membership/create.go

```diff
@@ -56,6 +56,17 @@ func create(w http.ResponseWriter, r *http.Request) {
 	}
 
 	tx := model.DB.Begin()
+
+	// Check if the plan is not deleted
+	plan := model.Plan{}
+	plan.ID = membership.PlanID
+	if err = tx.Model(&model.Plan{}).First(&plan).Error; err != nil {
+		tx.Rollback()
+		loggerx.Error(err)
+		errorx.Render(w, errorx.Parser(errorx.CannotSaveChanges()))
+		return
+	}
+
```

Comment:
As plan id is a Foreign Key if it does not exists or deleted then Membership cannot be created. I don't think this query is required.

Reply:
But when we delete a plan, it does not actually delete the plan, we are doing soft delete so the deleted_at field is set. Hence, the foreign key constrain does not fail when we add some plan_id which is deleted previously.

Reply:
Are you sure about that? If not once check it and make changes accordingly.

Reply:
Yaa I checked it.... I came across this bug while testing only so I fixed it.